beacon-chain/state/fieldtrie/field_trie_test.go

```diff
@@ -16,9 +17,14 @@ import (
 
 func TestFieldTrie_NewTrie(t *testing.T) {
 	newState, _ := util.DeterministicGenesisState(t, 40)
+	roots := newState.BlockRoots()
+	blockRoots := make([][32]byte, len(roots))
+	for i, r := range roots {
+		blockRoots[i] = [32]byte(r)
+	}
 
-	// 5 represents the enum value of state roots
-	trie, err := fieldtrie.NewFieldTrie(types.FieldIndex(5), types.BasicArray, newState.StateRoots(), uint64(params.BeaconConfig().SlotsPerHistoricalRoot))
+	// 5 represents the enum value of block roots
+	trie, err := state_native.NewFieldTrie(types.FieldIndex(5), types.BasicArray, customtypes.BlockRoots(blockRoots), uint64(params.BeaconConfig().SlotsPerHistoricalRoot))
 	require.NoError(t, err)
 	root, err := stateutil.RootsArrayHashTreeRoot(newState.StateRoots(), uint64(params.BeaconConfig().SlotsPerHistoricalRoot))
```

Comment:
If you are changing the field you should also change which field is hashed, otherwise the test makes no sense

Reply:
fixed, I also removed hardcoded numbers in other tests